### PR TITLE
Update spec and explainer for SANACLAP and exclusion API.

### PR DESCRIPTION
**Pull request by skobes**
Update spec and explainer for SANACLAP and exclusion API.

Also remove specification of 20-adjustment limit, bounce suppression, and
exclusion of position:absolute with non-zero offset, and add explainer comments
about open questions (layout-affecting properties, opt-in API, unconditional
exclusion of position:absolute, remaining issues with feedback loops).

### CODE REVIEW

File: scroll-anchoring/explainer.md [hunk 1]
```diff
@@ -76,38 +77,153 @@ document or an element with scrollable overflow) is as follows:
...
+Initial testing revealed that scroll anchoring often performed undesired scroll
+adjustments in response to changes in the computed style of the anchor node or
+one of its ancestors.  For example, a map panning interface might update a CSS
+transform in response to mouse movements, or a sticky-header implementation
+might adjust padding on the body element.
+
+The SANACLAP principle ("suppress if anchor node ancestor changed a
+layout-affecting property") aims to suppress these undesired scroll anchoring
+adjustments.
+
```
Comment (KenjiBaheux) [inline scroll-anchoring/explainer.md]: Nit: maybe bold the letters used by the acronym as in "**s**uppress if **a**nchor **n**ode **a**ncestor **c**hanged a **l**ayout-**a**ffecting **p**roperty" ?

File: scroll-anchoring/explainer.md [hunk 2]
```diff
@@ -76,38 +77,153 @@ document or an element with scrollable overflow) is as follows:
...
+Under the SANACLAP principle, if the anchor node, or any ancestor of the anchor
+node up to and including the scrollable element (or document), has any of a
+certain set of CSS properties modified (the "layout-affecting properties"), any
+scroll anchoring adjustment that would have resulted from that modification is
+suppressed.
+
+The exact set of properties that are considered layout-affecting for the purpose
+of scroll anchoring is subject to change, but should at minimum include:
+
```
Comment (KenjiBaheux) [inline scroll-anchoring/explainer.md]: Is there any property that could be construed as layout-affecting but yet ought to be excluded (e.g. turned out to be a hassle in practice, or is excluded for logical reason) that would be worth calling out?

File: scroll-anchoring/explainer.md [hunk 3]
```diff
@@ -76,38 +77,153 @@ document or an element with scrollable overflow) is as follows:
...
+We propose a CSS property `overflow-anchor` whose values have the following
+meaning for some element E:
 
-* `overflow-anchor: visible` causes a scrollable area to follow the content
-  within its viewport, using the scroll anchoring algorithm described in this
-  proposal.
+* `overflow-anchor: visible` declares that the DOM subtree rooted at E is
+  eligible to participate in the anchor node selection algorithm for any
+  scrollable area created by E or an ancestor of E.
```
Comment (KenjiBaheux) [inline scroll-anchoring/explainer.md]: I was expecting `visible` to be paired with its antonym `hidden`, not `none`.
Will think about it during lunch.

File: scroll-anchoring/spec.html [hunk 4]
```diff
@@ -182,60 +188,54 @@
...
-              <ol>
-                <li>50 px downward
-                <li>50 px upward
-                <li>50 px downward
-              </ol>
-              The first two will produce scroll anchoring adjustments, but the
-              third will not.
+              When the DOM changes that caused the adjustment include changes to the
+              computed values of properties affecting the position of elements, on
+              elements in the path from the anchor node to the scrollable element (or
```
Comment (ymalik) [inline scroll-anchoring/spec.html]: We should probably be more clear about the "elements in the path"?  These are probably, "all" elements, or "aunts/uncles/siblings/cousins"?

File: scroll-anchoring/explainer.md (hunk 1 above)
Comment (skobes) [inline scroll-anchoring/explainer.md]: Done.

File: scroll-anchoring/explainer.md (hunk 2 above)
Comment (skobes) [inline scroll-anchoring/explainer.md]: This is something we still need to work out the details of.  I added some comments about it to the Open Issues section of the explainer (under "Definition of Layout-Affecting Property").

The ones I listed in the SANACLAP section are the ones I consider the most important.  The [spreadsheet](https://docs.google.com/spreadsheets/d/1iyg8udD-QNwv9sUja7vDBJI6MdKmFwczfvwwW2BAEzg/edit) has, in addition to these, a lot of text rendering properties, and a lot of really obscure stuff, that I think are not important for compat.

File: scroll-anchoring/explainer.md (hunk 3 above)
Comment (skobes) [inline scroll-anchoring/explainer.md]: Yeah these names are awkward.  I added an "issue" paragraph in the spec to come up with better names.  I'm hoping we can have that discussion with the WICG, along with the question of separating opt out and exclusion.

File: scroll-anchoring/spec.html (hunk 4 above)
Comment (skobes) [inline scroll-anchoring/spec.html]: Reworded to "any element in the path".

File: scroll-anchoring/explainer.md [hunk 5]
```diff
@@ -76,38 +77,161 @@ document or an element with scrollable overflow) is as follows:
...
+The current anchor node selection algorithm has no special treatment for
+absolute-positioned elements unless their containing block is outside the
+scroller.  (An absolute-positioned element whose containing block is outside
+the scroller is not part of the scrolling content, so it is clearly futile to
+anchor to such an element.)
+
+There is an open question of whether to skip absolute-positioned elements
+unconditionally.
+
+The argument in favor is that an absolute-positioned element is not affected by
```
Comment (ojanvafai) [inline scroll-anchoring/explainer.md]: IMO, this is the principled argument. The pragmatic argument is that we'll successfully scroll anchor on pages that have a toolbar at the top of the page that is absolutely positioned.

Comment (skobes) [inline scroll-anchoring/explainer.md]: Right, I forgot about that argument.  Added.

**Comment by ojanvafai** [comment]
lgtm

**Pull request merged**